src/main/java/trivia/game/GameBetter.java

```diff
@@ -0,0 +1,111 @@
+    Board board = new Board();
+    PlayerManagement players = new PlayerManagement();
+
+    public GameBetter() {
+        this.questions.generateRandom(QUESTIONS_NUMBER);
+    }
+
+    public boolean add(String playerName) {
+        players.addOne(new Player(playerName));
+        log(playerName + " was added");
+        log("They are player number " + players.totalNumber());
+        return true;
+    }
+
+    public void roll(int roll) {
+        if (!isPlayable()) {
+            log("Not enough players to start the game!");
+            return;
+        }
+        play(roll);
+    }
+
+    public boolean wasCorrectlyAnswered() {
+        Player currentPlayer = players.getCurrentPlayer();
+        boolean gameFinished;
+
+        if (jail.hasImprisoned(currentPlayer)) {
+            return checkIfIsGettingOut(currentPlayer);
+        } else {
+            log("Answer was correct!!!!");
+            currentPlayer.addCoins();
+            log(currentPlayer.getName() + " now has " + currentPlayer.getCoins() + " Gold Coins.");
+            gameFinished = !isGameFinished(currentPlayer);
+            players.nextPlayer();
+            return gameFinished;
+        }
+    }
+
+    public boolean wrongAnswer() {
+        Player currentPlayer = players.getCurrentPlayer();
+        log("Question was incorrectly answered");
+        jail.addPrisoner(currentPlayer);
+        players.nextPlayer();
+        return true;
+    }
+
+    private boolean isGameFinished(Player currentPlayer) {
+        return currentPlayer.getCoins() == WINNING_COINS;
+    }
+
+    private void askQuestion(QuestionCategory category) {
+        log("The category is " + category.getLabel());
+        log(questions.showNextQuestion(category));
+    }
+
+    private boolean isPlayable() {
+        return players.totalNumber() >= 2;
+    }
+
+    private boolean checkIfIsGettingOut(Player player) {
+        if (jail.isPlayerIsGettingOut()) {
+            log("Answer was correct!!!!");
+            jail.removePrisoner(player);
+            player.addCoins();
+            log(player.getName() + " now has " + player.getCoins() + " Gold Coins.");
+            boolean gameFinished = !isGameFinished(player);
+            players.nextPlayer();
+            return gameFinished;
+        } else {
+            players.nextPlayer();
+            return true;
+        }
+    }
+
+    private void play(int roll) {
+
+        Player currentPlayer = players.getCurrentPlayer();
+        log(currentPlayer.getName() + " is the current player");
+        log("They have rolled a " + roll);
+
+        if (!jail.hasImprisoned(currentPlayer)) {
+            nextStep(currentPlayer, roll);
+            return;
+        }
+        if (jail.tryToGetOut(currentPlayer, roll)) {
+            log(currentPlayer.getName() + " is getting out of the penalty box");
+            nextStep(currentPlayer, roll);
+        }
+    }
+
+    private void nextStep(Player player, int roll) {
+        player.move(roll);
```

Comment:
move and checkPosition belong to the same method, because it's part of the 'move' logic to actually know when player is at the end of the round. And again, check position doesn't actually check anything, it is a command.